app/models/text_component.rb

```diff
@@ -3,7 +3,9 @@ class TextComponent < ActiveRecord::Base
   has_and_belongs_to_many :triggers
   has_many :sensors, through: :triggers
   has_many :events, through: :triggers
+  has_and_belongs_to_many :channels
   belongs_to :report
+  belongs_to :publishing_context
```

Comment:
This is the keyword that the bot later will be triggered on. It makes way more sense to have a text component defining be under a keyword (context) than to tie the keyword to a channel.